src/ModelContextProtocol/Server/AIFunctionMcpServerTool.cs

```diff
@@ -195,57 +195,59 @@ public override async Task<CallToolResponse> InvokeAsync(
             };
         }
 
-        switch (result)
+        return result switch
         {
-            case null:
-                return new()
-                {
-                    Content = []
-                };
-
-            case string text:
-                return new()
-                {
-                    Content = [new() { Text = text, Type = "text" }]
-                };
-
-            case TextContent textContent:
-                return new()
-                {
-                    Content = [new() { Text = textContent.Text, Type = "text" }]
-                };
-
-            case DataContent dataContent:
-                return new()
-                {
-                    Content = [new()
+            null => new()
+            {
+                Content = []
+            },
```

Comment:
As long as you're here, can you also special-case Content:
```C#
Content content => new() { Content = [content] }
```
and
```C#
IEnumerable<Content> contents => new() { Content = contents.ToList() }
```
or something like that.